### PR TITLE
Add shape existence check in GatherElements shape inference logic

### PR DESCRIPTION
**Pull request by hariharans29**
The shape inference logic basically propagated the shape of the `indices` input to the output shape. This logic is correct but we need to add a guard to make sure the `indices` input has a shape associated with it.

### CODE REVIEW

File: onnx/test/shape_inference_test.py [hunk 1]
```diff
@@ -2707,6 +2707,22 @@ def test_pad(self):  # type: () -> None
...
+            [])
+        self._assert_inferred(graph, [make_tensor_value_info('y', TensorProto.FLOAT, (2,))])
+
+    def test_gatherelements_noshape(self):  # type: () -> None
+        graph = self._make_graph(
+            [('x', TensorProto.FLOAT, (6,)),
+             ('indices', TensorProto.INT64, ())],
+            [make_node('GatherElements', ['x', 'indices'], ['y'])],
+            [])
+        self._assert_inferred(graph, [make_tensor_value_info('y', TensorProto.FLOAT, ())])
```
Comment (skottmckay) [inline onnx/test/shape_inference_test.py]: Should this be
('indices', TensorProto.INT64, None)],
and
make_tensor_value_info('y', TensorProto.FLOAT, None)?

File: onnx/defs/tensor/defs.cc [hunk 2]
```diff
@@ -1227,7 +1227,10 @@ ONNX_OPERATOR_SET_SCHEMA(
             "Constrain indices to integer types")
         .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
           propagateElemTypeFromInputToOutput(ctx, 0, 0);
-          propagateShapeFromInputToOutput(ctx, 1, 0);
+          // propogate indices' shape to output if it exists
```
Comment (skottmckay) [inline onnx/defs/tensor/defs.cc]: nit: propagate

File: onnx/test/shape_inference_test.py (hunk 1 above)
Comment (hariharans29) [reply onnx/test/shape_inference_test.py]: You are right. Fixed, thanks.

File: onnx/defs/tensor/defs.cc (hunk 2 above)
Comment (hariharans29) [reply onnx/defs/tensor/defs.cc]: Fixed, thanks

**Pull request merged**